### ISSUE TITLE
EIGEN_MAKE_UNALIGNED_ARRAY_ASSERT

### ISSUE THREAD

**Issue by TeamSegFault** [dilevin/computer-graphics-meshes#10]
Hi,

EIGEN_MAKE_UNALIGNED_ARRAY_ASSERT is raised for me when QuadViewer() is called. Below is the error message:

Assertion failed: (reinterpret_cast<size_t>(array) & 0xf) == 0 && "this assertion is explained here: " "http://eigen.tuxfamily.org/dox-devel/group__TopicUnalignedArrayAssert.html" " **** READ THIS WEB PAGE !!! ****"

Read the webpage but since the assert is raised in QuadViewer() I can't pinpoint why. Any ideas why this assert is raised?

**Comment by WChrisK** [comment]
What did the backtrace say when you ran it? As per instructions at the top:

```bash
$ gdb ./obj
> run
> bt
```

We need more info to help you (unless someone else happened to have this exact problem)

**Comment by TeamSegFault** [comment]
Right, I'm not using GDB per se but here's the call stack from VS. Essentially, in main(), QuadViewer object is constructed then the assert is raised. I think QuadViewer displays the meshes from the matrices we pass but not sure why the assert would be thrown.

![image](https://user-images.githubusercontent.com/45057012/53220004-50984f80-3630-11e9-84ad-7e852159f3fd.png)

**Comment by WChrisK** [comment]
That is interesting. I tried looking into it myself and couldn't find any reason why it'd do that. I should try it on VS2017 tomorrow and see if that happens to me as well (are you using VS2017?)

I have something that may help, but if it doesn't work then I'm out of ideas :(

You might be able to add the following into your CMakeLists.txt and hope that it's carried over into your .sln/.proj files, assuming I did it right:

Put these on line 5 in `CMakeLists.txt`:
```
add_definitions(-DEIGEN_DONT_VECTORIZE -DEIGEN_DISABLE_UNALIGNED_ARRAY_ASSERT)
add_compile_definitions(EIGEN_MAX_STATIC_ALIGN_BYTES=0)
```
and hope it compiles!

If it does compile, you are probably good to go since when it gets tested on CDF it should work anyways. I do not know if the above however will lead to cryptic crashes or anything weird... it hopefully doesn't but I am not familiar with Eigen enough and only know the basics of SIMD, so me trying to imply it won't explode violently would be a mistake on my part.
